newsfragments/2292.bugfix

```diff
@@ -0,0 +1 @@
+Pin boost to 1.74
```

Comment:
Add comment linking to https://github.com/boostorg/geometry/issues/590 for background?